test/network/network.node.js

```diff
@@ -199,4 +197,28 @@ describe('network', () => {
     await networkA.sendMessage(p2pC.peerInfo.id, msg)
     await deferred.promise
   })
+
+  it('dials to peer using Bitswap 1.2.0', async () => {
+    networkA = new Network(p2pA, bitswapMockA)
+
+    // only supports 1.2.0
+    networkB = new Network(p2pB, bitswapMockB)
+    networkB.protocols = ['/ipfs/bitswap/1.2.0']
+
+    networkA.start()
+    networkB.start()
+
+    // FIXME: have to already be connected as sendMessage only accepts a peer id, not a PeerInfo
```

Comment:
This gets fixed in #217

Reply:
Could you create an issue to track this please?

Reply:
I already updated #217 to add the peer multiaddrs to the AddressBook before the `sendMessage` and removed the dial operation from the test.

We are removing the PeerInfo from libp2p@0.28. `multiaddrs` for the peer should be in the AddressBook before any attempt to dial using a PeerId. Considering this case, the nodes needed to "discover" themselves, in order to send a message. The "discover" here is adding the multiaddrs to the AddressBook and then use the `sendMessage` with the PeerId.

So, the goal here is to use the new libp2p and not to accept PeerInfo in the sendMessage. I can create an issue to track this, but I am considering this as part of updating libp2p in bitswap.

```js
it('dials to peer using Bitswap 1.2.0', async () => {
    networkA = new Network(p2pA, bitswapMockA)

    // only supports 1.2.0
    networkB = new Network(p2pB, bitswapMockB)
    networkB.protocols = ['/ipfs/bitswap/1.2.0']

    networkA.start()
    networkB.start()

    const deferred = pDefer()

    bitswapMockB._receiveMessage = () => {
      deferred.resolve()
    }

    p2pA.peerStore.addressBook.set(p2pB.peerId, p2pB.multiaddrs)
    await networkA.sendMessage(p2pB.peerId, new Message(true))

    return deferred
  })
```